workos/utils/_base_http_client.py

```diff
@@ -66,14 +66,13 @@ def __init__(
         timeout: Optional[int] = DEFAULT_REQUEST_TIMEOUT,
     ) -> None:
         self._api_key = api_key
-        # Template for constructing the URL for an API request
-        self._base_url = "{}{{}}".format(base_url)
+        self._base_url = base_url
```

Comment:
This was the real culprit of the mixup. `base_url` was `https://api.workos.com/{}` here, but only `https://api.workos.com/` in ClientConfiguration. I felt like `base_url` should just be the URL itself, so moved the formatting below.